app/services/liquid_cached_assigns.rb

```diff
@@ -0,0 +1,57 @@
+class LiquidCachedAssigns < Hash
+  AVAILABLE_PROJECT_ASSIGNS = %w{ evidences issues nodes notes project tags }.freeze
+
+  attr_accessor :assigns, :project
+
+  def initialize(project:)
+    @project = project
+
+    @assigns = { 'project' => ProjectDrop.new(project) }
+    @assigns.merge!(assigns_pro)
+  end
+
+  def [](record_type)
+    assigns[record_type] ||= cached_drops(record_type)
+  end
+
+  def key?(key)
```

Comment:
I'm assuming this is needed on the Liquid side of things?

Reply:
Yes. This is used here: https://github.com/Shopify/liquid/blob/77bc56a1c28a707c2b222559ffb0b7b1c5588928/lib/liquid/context.rb#L211. Liquid checks if the assign is present in the hash, but since we're lazily loading the assigns we need to return the available assigns even if they're not yet part of the hash.

Reply:
Added comment for future reference